arch/arm64: Support runtime frequency

hx-bora opened the pull request:
Add ARM64 support for `CONFIG_TIMER_READS_ITS_FREQUENCY_AT_RUNTIME`.

Used for the `qemu_kvm_arm64` board. This should have been part of #51932. Without this, the `qemu_kvm_arm64` board is painfully slow due to the timer frequency being wrong.

carlocaione commented on drivers/timer/arm_arch_timer.c:
```diff
@@ -204,6 +204,18 @@ static int sys_clock_driver_init(const struct device *dev)
 	irq_enable(ARM_ARCH_TIMER_IRQ);
 	arm_arch_timer_set_irq_mask(false);
 
+#ifdef CONFIG_TIMER_READS_ITS_FREQUENCY_AT_RUNTIME
+#ifdef CONFIG_ARM64
+	extern int z_clock_hw_cycles_per_sec;
+	uint64_t cntfrq_el0 = read_cntfrq_el0();
+
+	__ASSERT(cntfrq_el0 < INT_MAX, "cntfrq_el0 cannot fit in system 'int'");
+	z_clock_hw_cycles_per_sec = (int) cntfrq_el0;
+#else
+#warning 'TIMER_READS_ITS_FREQUENCY_AT_RUNTIME' is not supported in arm_arch_timer
+#endif
+#endif
```
Can we move everything in `arm_arch_timer_init()` in `include/zephyr/arch/arm64/timer.h` instead?

Commit:
arch/arm64: Support runtime frequency

Add ARM64 support for `CONFIG_TIMER_READS_ITS_FREQUENCY_AT_RUNTIME`.

Used for the `qemu_kvm_arm64` board. This should have been part
of #51932. Without this, the `qemu_kvm_arm64` board is painfully
slow due to the timer frequency being wrong.

Signed-off-by: Henri Xavier <datacomos@huawei.com>
